### PR TITLE
feat: UI/UX improvements and small fixes

### PR DESCRIPTION
**Pull request by pedroferreira1**
### Acceptance Criteria
- A list of bug fixes and small UI/UX improvements listed below.

1. Don't show the option to "Always show token" inside Transaction Detail > Register token modal because it will open modal over modal (see video below).

image

https://github.com/HathorNetwork/hathor-wallet/commit/c68671526931aa59942134af8ed526feb7bf19b6

2. Click on label works for checkbox click as well in the AddToken modal for Always Show Token.

https://github.com/HathorNetwork/hathor-wallet/commit/694adef08640070830b1be18c5864bfc0c5a7b51

3. If you mark to hide zero balance tokens and the token currently selected has zero balance, then you must select HTR.

https://github.com/HathorNetwork/hathor-wallet/commit/4cdb89cf517acdf6ad7fe5237a1d3e6a1e3ab34a

4. The first time you open the app and load a wallet, the progress goes to 0 when loading before changing to the Wallet screen (see video below).

https://user-images.githubusercontent.com/3298774/172913260-0ea9149d-e403-498d-87c9-a6fa543d000e.mov

https://github.com/HathorNetwork/hathor-wallet/commit/ceb48b74bc1e3e8ad3836b32a75bf9af962b35fc

6. Newly registered/created tokens were not showing an Administrative Tab in the main Wallet screen because we should call the method in the component did mount as well.

https://github.com/HathorNetwork/hathor-wallet/commit/c775807152dcb0da1fd6ddb5f83920aa94a90a38

7. New created tokens were not being added to the `allTokens` redux set, which generated a bug if you create a token and unregister it (it wouldn't show up in the purple circle number).

https://github.com/HathorNetwork/hathor-wallet/commit/e359322a0d0e5077969fa9ebb73b98391a0cea6a

8. The Unknown Tokens screen was broken if you had there a token without any transactions (you wouldn't have this token in the `tokensBalance` object).

https://github.com/HathorNetwork/hathor-wallet/commit/8e57c941e0851dea1f8d5d7d4b18dc09fb2c281f

### Security Checklist
- [x] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/reducers/index.js [hunk 1]
```diff
@@ -421,4 +423,25 @@ export const onResolveLockWalletPromise = (state, action) => {
   }
 };
 
+/*
+ * Used When we select to hide zero balance tokens and a token with zero balance is selected
+ * In that case we must select HTR
+*/
+export const resetSelectedTokenIfNeeded = (state, action) => {
+  const tokensBalance = state.tokensBalance;
+  const selectedToken = state.selectedToken
```
Comment (tuliomir) [inline src/reducers/index.js:432]:
```diff
-  const selectedToken = state.selectedToken
+  const selectedToken = state.selectedToken;
```

Comment (pedroferreira1) [reply src/reducers/index.js]: Done https://github.com/HathorNetwork/hathor-wallet/pull/321/commits/5cbb43455dc639f2b49f76c448cbf940dd4027fa

File: src/reducers/index.js [hunk 2]
```diff
@@ -421,4 +423,36 @@ export const onResolveLockWalletPromise = (state, action) => {
...
+    return {
+      ...state,
+      selectedToken: hathorLib.constants.HATHOR_TOKEN_CONFIG.uid
+    };
+  }
+
+  return state;
+};
+
+export const onNewTokens = (state, action) => {
```
Comment (r4mmer) [inline src/reducers/index.js:447]:
```suggestion
/*
 * Used when registering or creating tokens to update the wallet token list.
*/
export const onNewTokens = (state, action) => {
```

**Review by r4mmer — APPROVED** [review]

File: src/reducers/index.js (hunk 2 above)
Comment (pedroferreira1) [reply src/reducers/index.js:450]: Done https://github.com/HathorNetwork/hathor-wallet/pull/321/commits/7eecb6bccf71c0372710636aab6b6210b3e2abef

**Pull request merged**